### PR TITLE
For getCompletionsAtPosition, require a flag to provide completions with code actions

### PR DESCRIPTION
**Pull request by andy-ms**

### CODE REVIEW

File: src/server/protocol.ts [hunk 1]
```diff
@@ -1617,6 +1617,7 @@ namespace ts.server.protocol {
          * Optional prefix to apply to possible completions.
          */
         prefix?: string;
+        includeCompletionsWithActions: boolean;
```
Comment (mhegazy) [inline src/server/protocol.ts]: i would rather we be more specific about what the feature does today, and make it `includeNonImportedDeclarations` or something along these lines.

Also please add a comment explaining what it does

**Comment by mhegazy** [comment]
since this is a breaking change, we should rev the API version.

File: src/server/protocol.ts (hunk 1 above)
Comment (andy-ms) [reply src/server/protocol.ts]: How about `includeExternalModuleExports`?

Comment (mhegazy) [reply src/server/protocol.ts]: 👍

**Comment by andy-ms** [comment]
@mhegazy Good to go?

**Comment by mjbvz** [comment]
Will add this flag to vs code for 1.18

File: src/server/protocol.ts [hunk 2]
```diff
@@ -1617,6 +1617,7 @@ namespace ts.server.protocol {
          * Optional prefix to apply to possible completions.
          */
         prefix?: string;
+        includeExternalModuleExports: boolean;
```
Comment (mhegazy) [inline src/server/protocol.ts]: please add a comment

**Comment by mhegazy** [comment]
@amcasey can you take this one for a spin as well.

**Comment by andy-ms** [comment]
Waiting for approval from @amcasey

**Comment by mhegazy** [comment]
Please port to release-2.6 as well.

File: src/services/types.ts [hunk 3]
```diff
@@ -310,6 +310,10 @@ namespace ts {
         dispose(): void;
     }
 
+    export interface GetCompletionsAtPositionOptions {
+        includeExternalModuleExports: boolean;
```
Comment (amcasey) [inline src/services/types.ts]: This name suggests we're going to have a separate opt-in for each kind of completion that might return an action.  Is that the plan?

Comment (mhegazy) [reply src/services/types.ts]: I could not think of another feature that we would add, and thus did not have a frame of reference. We also got feedback form the VSCode team that they would like to be able to disable the feature altogether, and incidentally from [customer reports](https://github.com/Microsoft/TypeScript/issues/19764) later on.

**Pull request merged**